src/mumble/MainWindow.cpp

```diff
@@ -534,12 +539,16 @@ void MainWindow::closeEvent(QCloseEvent *e) {
 			return;
 		}
 
-		if (qcbRemember->isChecked()) {
-			Global::get().s.connectedQuitBehavior = ConnectedQuitBehavior::CLOSE;
+		/* If checkbox is checked and not connected, keep asking when connected
+		 * If checkbox is checked and connected, quit always */
+		if (qcbRemember->isChecked() && !(sh && sh->isRunning())) {
+			Global::get().s.quitBehavior = QuitBehavior::ASK_WHEN_CONNECTED;
```

Comment:
Here, the logic works out, but in order to be consistent with the minimize option, I think I would opt for `ALWAYS_QUIT` here :thinking: